hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/AbstractJobProxy.java

```diff
@@ -306,4 +311,37 @@ private void rejoinJob(Throwable t) {
             future.internalCompleteExceptionally(new CancellationException(msg));
         }
     }
+
+    static final class JobMetricsImpl implements JobMetrics {
+
+        private final Map<String, Long> metrics;
+
+        private JobMetricsImpl(Map<String, Long> metrics) {
+            this.metrics = Collections.unmodifiableMap(new HashMap<>(metrics));
+        }
+
+        static JobMetrics of(Map<String, Long> metrics) {
+            return new JobMetricsImpl(metrics);
+        }
+
+        @Nonnull
+        @Override
+        public Collection<String> getMetricNames() {
+            return metrics.keySet();
+        }
+
+        @Override
+        public Long getMetricValue(String name) throws IllegalArgumentException {
+            if (name == null) {
+                throw new IllegalArgumentException("Metric name can't be null!");
+            }
+
+            Long value = metrics.get(name);
+            if (value == null) {
+                throw new IllegalArgumentException("Metric with name '" + name + "' not available for job!");
```

Comment:
it's better to return null here than throw

Reply:
Ok, but then I can't have a primitive "long" return type (just saying I can't comply to both requests).